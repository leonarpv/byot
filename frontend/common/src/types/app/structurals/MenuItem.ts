import * as React from 'react';

export interface MenuItem<I = string | number> {
  label: string;
  id: I;
  icon?: React.ReactNode;
}

export interface LinkMenuItem<I = string | number> extends MenuItem<I> {
  link: string;
}
