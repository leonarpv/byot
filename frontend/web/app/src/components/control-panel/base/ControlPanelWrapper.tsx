import * as React from 'react';
import {makeStyles, Theme} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {ControlPanelSidebar} from './ControlPanelSidebar';
import {LinkMenuItem} from '@byot-frontend/common/src/types/app/structurals/MenuItem';

interface Props extends WithStyles<typeof styles> {
  children: React.ReactNode;
  menuItems: LinkMenuItem[];
  selected?: string;
}

const styles = (theme: Theme) => ({
  content: {
    [theme.breakpoints.between('sm', 'md')]: {
      paddingLeft: '33%',
    },
    [theme.breakpoints.up('md')]: {
      paddingLeft: '25%',
    },
  },
});
const useStyles = makeStyles(styles);

export const ControlPanelWrapper: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <>
      <ControlPanelSidebar selected={props.selected} menuItems={props.menuItems} />
      <div className={styles.content}>{props.children}</div>
    </>
  );
};
