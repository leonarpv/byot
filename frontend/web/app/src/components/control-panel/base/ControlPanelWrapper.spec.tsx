import * as React from 'react';
import {render} from '@testing-library/react';
import {ControlPanelWrapper} from './ControlPanelWrapper';
import {LinkMenuItem} from '@byot-frontend/common/src/types/app/structurals/MenuItem';

describe('<ControlPanelWrapper/>', () => {
  it('should render', () => {
    const {container} = render(<ControlPanelWrapper menuItems={[]}>{''}</ControlPanelWrapper>);
    expect(container).toMatchSnapshot();
  });
  it('should render as many items as provided + logout item', () => {
    const menuItems: LinkMenuItem[] = [
      {
        id: 0,
        label: 'Label 0',
        link: '/route-0',
      },
      {
        id: 1,
        label: 'Label 1',
        link: '/route-1',
      },
    ];
    const {container, queryAllByTestId} = render(
      <ControlPanelWrapper menuItems={menuItems}>{''}</ControlPanelWrapper>
    );
    expect(queryAllByTestId('control-panel-drawer-menu-item')).toHaveLength(menuItems.length + 1);
    expect(container).toMatchSnapshot();
  });
});
