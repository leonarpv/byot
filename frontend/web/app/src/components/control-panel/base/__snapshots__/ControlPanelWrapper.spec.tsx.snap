// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlPanelWrapper/> should render 1`] = `
<div>
  <div
    class="makeStyles-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-6 makeStyles-root-3 makeStyles-root-5"
    >
      <img
        class="makeStyles-img-4"
        src="logo.svg"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-root-7"
    >
      <ul
        class="MuiList-root"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-9"
    >
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-root-7"
      >
        <ul
          class="MuiList-root"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-11 MuiListItem-gutters MuiListItem-button"
            data-testid="control-panel-drawer-menu-item"
            role="button"
            tabindex="0"
            to="/logout"
          >
            <span
              class="makeStyles-highlight-12 makeStyles-highlight-15"
            />
            <div
              class="MuiListItemIcon-root makeStyles-icon-13 makeStyles-icon-16"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10.79 16.29c.39.39 1.02.39 1.41 0l3.59-3.59c.39-.39.39-1.02 0-1.41L12.2 7.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L12.67 11H4c-.55 0-1 .45-1 1s.45 1 1 1h8.67l-1.88 1.88c-.39.39-.38 1.03 0 1.41zM19 3H5c-1.11 0-2 .9-2 2v3c0 .55.45 1 1 1s1-.45 1-1V6c0-.55.45-1 1-1h12c.55 0 1 .45 1 1v12c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1v3c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-14 makeStyles-textPrimary-17 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Log out
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <p
        class="MuiTypography-root makeStyles-typography-18 MuiTypography-body1"
      >
        All rights reserved to
        <strong>
           byoT
        </strong>
      </p>
    </div>
  </div>
  <div
    class="makeStyles-content-1"
  />
</div>
`;

exports[`<ControlPanelWrapper/> should render as many items as provided + logout item 1`] = `
<div>
  <div
    class="makeStyles-root-20"
  >
    <div
      class="MuiBox-root MuiBox-root-24 makeStyles-root-21 makeStyles-root-23"
    >
      <img
        class="makeStyles-img-22"
        src="logo.svg"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-26 makeStyles-root-25"
    >
      <ul
        class="MuiList-root"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-root-27 MuiListItem-gutters MuiListItem-button"
          data-testid="control-panel-drawer-menu-item"
          role="button"
          tabindex="0"
          to="/route-0"
        >
          <span
            class="makeStyles-highlight-28 makeStyles-highlight-31"
          />
          <div
            class="MuiListItemIcon-root makeStyles-icon-29 makeStyles-icon-32"
          />
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-30 makeStyles-textPrimary-33 MuiTypography-body1 MuiTypography-displayBlock"
            >
              Label 0
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-root-27 MuiListItem-gutters MuiListItem-button"
          data-testid="control-panel-drawer-menu-item"
          role="button"
          tabindex="0"
          to="/route-1"
        >
          <span
            class="makeStyles-highlight-28 makeStyles-highlight-34"
          />
          <div
            class="MuiListItemIcon-root makeStyles-icon-29 makeStyles-icon-35"
          />
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-30 makeStyles-textPrimary-36 MuiTypography-body1 MuiTypography-displayBlock"
            >
              Label 1
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
    <div
      class="MuiBox-root MuiBox-root-37"
    >
      <div
        class="MuiBox-root MuiBox-root-38 makeStyles-root-25"
      >
        <ul
          class="MuiList-root"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-27 MuiListItem-gutters MuiListItem-button"
            data-testid="control-panel-drawer-menu-item"
            role="button"
            tabindex="0"
            to="/logout"
          >
            <span
              class="makeStyles-highlight-28 makeStyles-highlight-39"
            />
            <div
              class="MuiListItemIcon-root makeStyles-icon-29 makeStyles-icon-40"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10.79 16.29c.39.39 1.02.39 1.41 0l3.59-3.59c.39-.39.39-1.02 0-1.41L12.2 7.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L12.67 11H4c-.55 0-1 .45-1 1s.45 1 1 1h8.67l-1.88 1.88c-.39.39-.38 1.03 0 1.41zM19 3H5c-1.11 0-2 .9-2 2v3c0 .55.45 1 1 1s1-.45 1-1V6c0-.55.45-1 1-1h12c.55 0 1 .45 1 1v12c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1v3c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-30 makeStyles-textPrimary-41 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Log out
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <p
        class="MuiTypography-root makeStyles-typography-42 MuiTypography-body1"
      >
        All rights reserved to
        <strong>
           byoT
        </strong>
      </p>
    </div>
  </div>
  <div
    class="makeStyles-content-19"
  />
</div>
`;
